ocs_ci/ocs/replica_one.py

```diff
@@ -0,0 +1,288 @@
+            replica1_osd_deployment.append(deployment["metadata"]["name"])
+
+    return replica1_osd_deployment
+
+
+def scaledown_deployment(deployments_name: list[str]) -> None:
+    """
+    Scale down deployments to 0
+
+    Args:
+        deployments_name (list[str]): list of deployment names.
+
+    """
+    log.info("Starts Scaledown deployments")
+    deployment_obj = OCP(kind=DEPLOYMENT, namespace="openshift-storage")
+    for deployment in deployments_name:
+        deployment_obj.exec_oc_cmd(f"scale deployment {deployment} --replicas=0")
+        log.info(f"scaling to 0: {deployment}")
+
+
+def count_osd_pods() -> int:
+    """
+    Gets OSDs count in a cluster
+
+    Returns:
+        int: number of OSDs in cluster
+
+    """
+    return len(get_pods_having_label(label=OSD_APP_LABEL))
+
+
+def delete_replica_1_sc() -> None:
+    """
+    Deletes storage class associated with replica1
+
+    """
+    sc_obj = OCP(kind=STORAGECLASS, resource_name=REPLICA1_STORAGECLASS)
+    try:
+        sc_obj.delete(resource_name=REPLICA1_STORAGECLASS)
+    except CommandFailed as e:
+        if "Error is Error from server (NotFound)" in str(e):
+            log.info(
+                f"{REPLICA1_STORAGECLASS} not found, assuming it was already deleted"
+            )
+        else:
+            raise CommandFailed(f"Failed to delete storage class: {str(e)}")
+
+
+def purge_replica1_osd():
+    """
+    Purge OSDs associated with replica1
+        1. scale down its deployments to 0
+        2. use OSD removal template
+
+    """
+    deployments_name = get_replica1_osd_deployment()
+    log.info(f"Deployments Name: {deployments_name}")
+    scaledown_deployment(deployments_name)
+    replica1_osds = get_replica_1_osds()
+    log.info(f"OSDS : {replica1_osds.keys()}")
+    log.info(f"OSD IDs: {replica1_osds.values()}")
+    run_osd_removal_job(osd_ids=replica1_osds.values())
+
+
+def delete_replica1_cephblockpools_cr(cbp_object: OCP):
+    """
+    Deletes CR of cephblockpools associated with replica1
+
+    Args:
+        cbp_object (ocp.OCP): OCP object with kind=CEPHBLOCKPOOL
+
+    """
+    for i in range(0, len((cbp_object.data["items"]))):
+        cbp_cr_name = cbp_object.data["items"][i]["spec"]["deviceClass"]
+        log.info(f"cbp_cr_name: {cbp_cr_name}")
+        if cbp_cr_name in get_failure_domains():
+            log.info(f"Deleting {DEFAULT_CEPHBLOCKPOOL}-{cbp_cr_name}")
+            cbp_object.delete(resource_name=(f"{DEFAULT_CEPHBLOCKPOOL}-{cbp_cr_name}"))
+
+
+def modify_replica1_osd_count(new_osd_count):
+    """
+    Modify number of OSDs associated with replica1
+
+    Args:
+        new_osd_count (str): number, represent the duplicatoin number of replica1 osd.
+        for instance, selecting 2, creates 6 osds
+
+    """
+    storage_cluster = OCP(kind=STORAGECLUSTER, name="ocs-storagecluster")
+    storage_cluster.exec_oc_cmd(
+        f"patch storagecluster ocs-storagecluster -n openshift-storage "
```

Comment:
please replace the resource names with constants (they should already have ones saved in constants.py).

Referring to:
`ocs-storagecluster`
`openshift-storage`